core/src/test/java/com/google/cloud/sql/core/CoreSocketFactoryTest.java

```diff
@@ -133,8 +143,17 @@ public void setup() throws IOException, GeneralSecurityException, ExecutionExcep
 
     clientKeyPair = Futures.immediateFuture(new KeyPair(publicKey, privateKey));
 
+    defaultExecutor = CoreSocketFactory.getDefaultExecutor();
+
+    // Mock the API client for testing
     when(adminApi.instances()).thenReturn(adminApiInstances);
-    when(adminApiInstances.get(anyString(), anyString())).thenReturn(adminApiInstancesGet);
+
+    // Mock wrong project and instance
+    doThrow(fakeNotConfiguredException()).when(adminApiInstances).get(anyString(), anyString());
+    // Mock correct project, wrong instance
+    doThrow(fakeNotAuthorizedException()).when(adminApiInstances).get(eq("myProject"), anyString());
+    // Mock correct instance
+    doReturn(adminApiInstancesGet).when(adminApiInstances).get(eq("myProject"), eq("myInstance"));
```

Comment:
I also tried this as well, but it quits matching without the "eq" for whatever reason.

Reply:
That is surprising to me. The rule is that if any argument uses a matcher, they all must; if none do it implies matching using `equals(Object)`.

You don't use argument matchers everywhere. How come they are required here? Could it be that because you're stubbing the same method call with different matchers you can't use plain arguments on any of them?

Reply:
Looks like that's the case:

> If you are using argument matchers, all arguments have to be provided by matchers.
> The following example shows verification but the same applies to stubbing:
>
> ```java
>    verify(mock).someMethod(anyInt(), anyString(), eq("third argument"));
>    //above is correct - eq() is also an argument matcher
> 
>    verify(mock).someMethod(anyInt(), anyString(), "third argument");
>    //above is incorrect - exception will be thrown because third argument is given without an argument matcher. 
> ```